src/pages/playground/Decoder.tsx

```diff
@@ -0,0 +1,155 @@
+import { FSModule } from 'browserfs/dist/node/core/FS';
+import { Chapter, Variant } from 'js-slang/dist/types';
+import { decompressFromEncodedURIComponent } from 'lz-string';
+import { Dispatch } from 'react';
+import { AnyAction } from 'redux';
+import { getDefaultFilePath, getLanguageConfig } from 'src/commons/application/ApplicationTypes';
+import { overwriteFilesInWorkspace } from 'src/commons/fileSystem/utils';
+import {
+  showFullJSWarningOnUrlLoad,
+  showFulTSWarningOnUrlLoad,
+  showHTMLDisclaimer
+} from 'src/commons/utils/WarningDialogHelper';
+import {
+  addEditorTab,
+  removeEditorTabsForDirectory,
+  setFolderMode,
+  updateActiveEditorTabIndex
+} from 'src/commons/workspace/WorkspaceActions';
+import { WorkspaceLocation } from 'src/commons/workspace/WorkspaceTypes';
+import { playgroundConfigLanguage } from 'src/features/playground/PlaygroundActions';
+
+import { convertParamToBoolean, convertParamToInt } from '../../commons/utils/ParamParseHelper';
+import { IParsedQuery, parseQuery } from '../../commons/utils/QueryHelper';
+import { WORKSPACE_BASE_PATHS } from '../fileSystem/createInBrowserFileSystem';
+
+export type programConfig = {
```

Comment:
Types should be in `PascalCase`. Also, please use this instead of `object` in the control bar component.